flink-test-utils-parent/flink-table-filesystem-test-utils/src/test/java/org/apache/flink/table/file/testutils/TestFileSystemTableFactoryTest.java

```diff
@@ -64,4 +66,22 @@ void testCreateSourceSink() {
         DynamicTableSink sink = createTableSink(SCHEMA, options);
         assertThat(sink).isInstanceOf(FileSystemTableSink.class);
     }
+
+    @Test
+    void testCreateUnboundedSource() {
+        Map<String, String> options = new HashMap<>();
+        options.put(FactoryUtil.CONNECTOR.key(), "test-filesystem");
+        options.put("path", "/tmp");
+        options.put("format", "testcsv");
+        options.put("source.monitor-interval", "5S");
+
+        DynamicTableSource source = createTableSource(SCHEMA, options);
+        assertThat(source).isInstanceOf(TestFileSystemTableSource.class);
+
+        // assert source is unbounded when specify source.monitor-interval
+        ScanTableSource.ScanRuntimeProvider scanRuntimeProvider =
+                ((TestFileSystemTableSource) source)
+                        .getScanRuntimeProvider(ScanRuntimeProviderContext.INSTANCE);
+        assertThat(scanRuntimeProvider.isBounded()).isFalse();
```

Comment:
Can we add a new case where, when setting the monitor-interval and the execution mode is batch, it should be a bounded source?